### PR TITLE
[Tensor] Add Tensor member functions

### PR DESCRIPTION
**Pull request by djeong20**
This PR extends the current Tensor member functions.

**Changes proposed in this PR:**
- Add member functions to get tensor information.
- Implement skeleton code from #2378.

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test:   [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/tensor/tensor_v2.cpp [hunk 1]
```diff
@@ -55,4 +50,33 @@ void TensorV2::initialize(TensorBase::Initializer init) {
 
 void TensorV2::print(std::ostream &out) const { itensor->print(out); }
 
+void TensorV2::putData() const { itensor->putData(); }
+
+TensorBase::Initializer TensorV2::getInitializer() const {
```
[the remark this answers is not in the document]
Comment (djeong20) [reply nntrainer/tensor/tensor_v2.cpp:55]: I believe moving the Initializer to TensorV2 creates a circular dependency (TensorV2 uses TensorBase while TensorBase uses TensorV2::Initializer), which causes a problem. I think the only options we have would be keeping Initializer to TensorBase or making it as nntrainer::Initializer. Any thoughts?

File: nntrainer/tensor/tensor_v2.h [hunk 2]
```diff
@@ -122,6 +159,79 @@ class TensorV2 {
...
+  size_t channel() const;
+
+  /**
+   * @brief     return Tensor height size
+   * @retval    height size
+   */
+  size_t height() const;
+
+  /**
+   * @brief     return Tensor batch size
```
Comment (DonghakPark) [inline nntrainer/tensor/tensor_v2.h:230]: i think this is typo

File: nntrainer/tensor/tensor_v2.h [hunk 3]
```diff
@@ -122,6 +159,79 @@ class TensorV2 {
...
+  size_t bytes() const;
+
+  /**
+   * @brief     return Tensor batch size
+   * @retval    batch size
+   */
+  size_t batch() const;
+
+  /**
+   * @brief     return Tensor batch size
```
Comment (DonghakPark) [inline nntrainer/tensor/tensor_v2.h:218]: i think this is typo

**Review by DonghakPark — APPROVED** [review]
except typo, Good !!

File: nntrainer/tensor/tensor_v2.h (hunk 2 above)
Comment (djeong20) [reply nntrainer/tensor/tensor_v2.h]: fixed!

File: nntrainer/tensor/tensor_v2.h (hunk 3 above)
Comment (djeong20) [reply nntrainer/tensor/tensor_v2.h]: fixed! thanks :)

File: nntrainer/tensor/tensor_v2.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (djeong20) [reply nntrainer/tensor/tensor_v2.cpp]: Fixed `nntrainer::TensorBase::Initializer` to `nntrainer::Initializer`.

**Review by Hyoungsukim — APPROVED** [review]
lgtm

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**